q2_feature_table/_summarize/_visualizer.py

```diff
@@ -356,12 +361,46 @@ def _frequencies(table, axis):
     return pd.Series(data=table.sum(axis=axis), index=table.ids(axis=axis))
 
 
-def _frequency_summary(table, axis='sample'):
+def is_integer(x):
+    if isinstance(x, int):
+        return True
+    elif isinstance(x, float):
+        return x.is_integer()
+    else:
+        return False
+
+
+def _frequency_summary(table: biom.Table, axis='sample'):
     frequencies = _frequencies(table, axis=axis)
 
-    summary = pd.Series([frequencies.min(), frequencies.quantile(0.25),
-                         frequencies.median(), frequencies.quantile(0.75),
-                         frequencies.max(), frequencies.mean()],
+    table = table.to_dataframe()
```

Comment:
Here's a better approach we can use:
```python
nonzeros = table.matrix_data.data
```
This will take the BIOM table to a Sparse Matrix, and then the raw non-zero data, which will prevent setting up a dataframe and iterating over rows or columns, since we don't care about where the values _are_, just their intergery-ness.